test/functional/tests/data_integrity/test_data_integrity_5d.py

```diff
@@ -0,0 +1,206 @@
+            cache.stop()
+
+    with TestRun.step("Calculate md5 for each core"):
+        dev_md5s = [File(dev.full_path).md5sum() for dev in core_devices]
+
+    with TestRun.step("Compare md5 sums for cores and core devices"):
+        if core_md5s != dev_md5s:
+            TestRun.fail("MD5 sums of core and core device do not match!")
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core1", DiskTypeLowerThan("cache"))
+@pytest.mark.require_disk("core2", DiskTypeLowerThan("cache"))
+@pytest.mark.require_disk("core3", DiskTypeLowerThan("cache"))
+@pytest.mark.require_disk("core4", DiskTypeLowerThan("cache"))
+def test_data_integrity_5d():
+    """
+        title: |
+          Data integrity test on three cas instances with different
+          cache modes with duration time equal to 5 days
+        description: |
+          Create 3 cache instances with different cache modes on caches equal to 50GB
+          and cores equal to 150GB, and run workload with data verification.
+        pass_criteria:
+            - System does not crash.
+            - All operations complete successfully.
+            - Data consistency is preserved.
+    """
+    with TestRun.step("Prepare cache and core devices"):
+        cache_devices, core_devices = prepare_devices()
+
+    with TestRun.step("Run 4 cache instances in different cache modes, add single core to each"):
+        instances = [start_instance(cache_devices[0], core_devices[0], CacheMode.WT),
+                     start_instance(cache_devices[1], core_devices[1], CacheMode.WB),
+                     start_instance(cache_devices[2], core_devices[2], CacheMode.WA),
+                     start_instance(cache_devices[3], core_devices[3], CacheMode.WO)]
+        (caches, cores) = zip(*instances)
+
+    with TestRun.step("Run test workloads with verification"):
+        fio_run = Fio().create_command()
+        fio_run.io_engine(IoEngine.libaio)
+        fio_run.direct()
+        fio_run.time_based()
+        fio_run.do_verify()
+        fio_run.verify(VerifyMethod.md5)
+        fio_run.verify_dump()
+        fio_run.run_time(runtime)
+        fio_run.read_write(ReadWrite.randrw)
+        fio_run.io_depth(128)
+        fio_run.blocksize_range([(start_size, stop_size)])
+        for core in cores:
+            fio_job = fio_run.add_job()
+            fio_job.target(core)
+        fio_run.run()
+
+    with TestRun.step("Calculate md5 for each core"):
+        core_md5s = [File(core.full_path).md5sum() for core in cores]
+
+    with TestRun.step("Stop caches"):
+        for cache in caches:
+            cache.stop()
+
+    with TestRun.step("Calculate md5 for each core"):
+        dev_md5s = [File(dev.full_path).md5sum() for dev in core_devices]
+
+    with TestRun.step("Compare md5 sums for cores and core devices"):
+        if core_md5s != dev_md5s:
+            TestRun.fail("MD5 sums of core and core device do not match!")
```

Comment:
Added info about cache mode would be nice here as well

Reply:
Fixed.